### DISCUSSION TITLE
h3 tags rendered unusually small - odd css setting in a file I cannot locate

### DISCUSSION THREAD

**Discussion by solvworx** [PDConSec/vsc-print#356] [Q&A] [answered]
So at some point, third-level markdown headings (that would render to h3 tags) started rendering as extremely small (smaller than h6). And it's only h3. The others (h1,h2,h4,h5,h6) all render at normal sizes. When I invoke the print preview, bring up chrome dev tools, elements tab, select an h3, and look at styles, I see this:

![Screenshot 2024-11-16 at 15 03 15](https://github.com/user-attachments/assets/f511224c-e106-4af7-a5ea-82d465a90aaf)

When I de-select the font-size line, things are back to normal. Chrome dev-tools shows a filename as settings.css.
But when I switch to the sources tab and drill down to src/css/settings.css all I see is:

> Could not load content for webpack://./src/css/settings.css (Fetch through target failed: Unsupported URL scheme; Fallback: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME)

I looked in the GitHub repo for vsc-print, and the settings.css I found is apparently not the same one. So I'm lost as to ...

- where this file is
- how I can examine it
- what can be done to fix the problem

If this turns out to be a legitimate issue, I can open one, but I thought I'd go the discussion route first.

**Comment by PeterWone** [comment]
That file is bundled. You should be able to override it using a stylesheet of your own but I'll verify that before calling it advice

**Comment by PeterWone** [comment]
I just re-read your post

> I looked in the GitHub repo for vsc-print, and the settings.css I found is apparently not the same one. So I'm lost as to

It's not the same because it's partly generated from your VS Code settings. Which is why it's called `settings.css`.

A better question is why the settings file is being delivered for Markdown. It mostly applies to printing source code, not Markdown. This may be a consequence of my adding support for syntax colouring of source code in Markdown fenced blocks.

**Reply by solvworx** [reply]
Experimented with second machine that does not sync profiles, looks like it happens there also.

**Comment by PeterWone** [comment]
I appreciate your patience. At the moment my life is full of property sale prep and caring for my mum.

**Reply by solvworx** [reply]
No worries! I appreciate that vsc-print exists and is free. I hope things go well for you.

**Comment by PeterWone** [comment]
I've begun to investigate this and here's how it looks on my system.

![image](https://github.com/user-attachments/assets/321be0c2-6667-41c7-8bf8-c92b769ebcf8)

Now, you may recall me writing that `settings.css` is partly generated. The point of this is reflected in the name of the stylesheet: it's where I inject settings. When you print source code, the styling is slaved to your editor settings. Here's what happens when I set my editor font size and face to 20 "pixels" Courier New. Quotes because browser pixels aren't screen pixels. The old value was 14.

![image](https://github.com/user-attachments/assets/c4bba27a-1b1f-4bb2-bed8-67a6f19d6f58)

The font size has changed but the size is _not_ what I expected. So we have both a bug and a design issue. The bug is that the size isn't being correctly picked up, and the design issue is that the settings resource shouldn't be used at all by Markdown pages.

**Comment by PeterWone** [comment]
I tried adding a css file 

![image](https://github.com/user-attachments/assets/21b0370b-39ab-45d2-8cf5-8891d1d429f6)

but the link occurs before the link to the settings so it failed to override even when I added `!important`

I will see about suppressing inclusion of the settings link for Markdown but this may have consequences for the styling of source code in fenced blocks.

**Comment by PeterWone** [comment]
I was wondering why I would set h3 in settings. Git blame reveals this change came in with a PR a year ago, and why. When you print a group of files you can have the filepath shown as an h3 heading at the start of each file. This was to slave the heading size to the listing text. I'm modifying the class to `h3.filepath` and altering the heading generation to specify the class. That leaves the mystery of why size isn't changing with the editor setting (it also doesn't work for printing source).

**Comment by PeterWone** [comment]
And the answer is Print has its own setting for that. Blame says it was me but nobody can prove that!

**Comment by PeterWone** [comment]
Looking better.

![image](https://github.com/user-attachments/assets/d98260ae-a8b4-4cbc-91ee-c8e77006e89c)

**Comment by PeterWone** [comment]
Update your extension to 0.13.3 and let me know whether it's truly resolved by closing the issue or giving more details.

**Comment by solvworx** [comment, marked as answer]
Yes, 0.13.3 works. The last previous version that did not have the issue was 0.11.28, it seemed to start happening with 0.12.x. Thanks!
